tika-parsers/tika-parsers-classic/tika-parsers-classic-modules/tika-parser-zip-commons/src/main/java/org/apache/tika/detect/zip/DefaultZipContainerDetector.java

```diff
@@ -224,29 +230,53 @@ MediaType detectStreaming(InputStream input, Metadata metadata) throws IOExcepti
         } catch (Exception e) {
             //swallow
         }
-
+        
         return finalDetect(detectContext);
     }
-
-
+    
+    
     private MediaType detect(ZipArchiveEntry zae, ZipArchiveInputStream zis,
-                             StreamingDetectContext detectContext) throws IOException {
-        for (ZipContainerDetector d : zipDetectors) {
+            StreamingDetectContext detectContext) throws IOException {
+        for (ZipContainerDetector d : getDetectors()) {
             MediaType mt = d.streamingDetectUpdate(zae, zis, detectContext);
             if (mt != null) {
                 return mt;
             }
         }
         return null;
     }
-
+    
     private MediaType finalDetect(StreamingDetectContext detectContext) {
-        for (ZipContainerDetector d : zipDetectors) {
+        for (ZipContainerDetector d : getDetectors()) {
             MediaType mt = d.streamingDetectFinal(detectContext);
             if (mt != null) {
                 return mt;
             }
         }
         return MediaType.APPLICATION_ZIP;
     }
+    
+    private void setDetectors(List<ZipContainerDetector> zipDetectors) {
+        this.staticZipDetectors0 = zipDetectors;
+    }
+    
+    private List<ZipContainerDetector> getDetectors() {
+        ServiceLoader loader = getServiceLoader();
+        if (staticZipDetectors0 == null) {
+            staticZipDetectors0 = new ArrayList<ZipContainerDetector>();
+            staticZipDetectors0.addAll(loader.loadStaticServiceProviders(ZipContainerDetector.class));
+        }
+        List<ZipContainerDetector> zipDetectors = new ArrayList<>();
+        zipDetectors.addAll(staticZipDetectors0);
```

Comment:
DefaultDetector uses dynamic before static. This is the opposite order. I suggest to implement it the same way as DefaultDetector does.

Reply:
the comment in the file is "//TODO: OPCBased needs to be last!!!" - that's why I load the dynamic (OPCBased) last and not first. It seems to me to be a better solution than sorting the services somewhere else (as we discussed yesterday).
I don't know if there any good reason to load OPCBased last - could be that the OPCBased will "block" the other detectors (i.e. by always detecting something)

Reply:
That TODO comment is IMHO not a good argument to change that behavior in this particular case. Because every where else I saw that the dynamic services are used before the static once.
E.g. see service loader's method #loadServiceProviders()... see DefaultDetector...

Reply:
ok then we change it